lib/connection/connection_config.js

```diff
@@ -292,12 +294,23 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
       ErrorCodes.ERR_CONN_CREATE_INVALID_PRIVATE_KEY_PASS);
   }
 
-  const token = options.token;
+  let token = options.token;
   if (Util.exists(options.token)) {
     Errors.checkArgumentValid(Util.isString(token),
       ErrorCodes.ERR_CONN_CREATE_INVALID_OAUTH_TOKEN);
   }
 
+  if (authenticator === authenticationTypes.OAUTH_AUTHENTICATOR && !Util.string.isNotNullOrEmpty(token)) {
+    const tokenFilePath = options.token_file_path ? options.token_file_path : '/snowflake/session/token';
+    validateOnlyUserReadWritePermission(tokenFilePath);
```

Comment:
should we await here?

Reply:
it isn't async at all. I added promise then/catch here.

Reply:
I don't see the promise and the catch but:
```
(async () => await validateOnlyUserReadWritePermission(tokenFilePath))()
```
Are you sure that it will perform the validation and wait?
should not we write the code like 
```
validateOnlyUserReadWritePermission(tokenFilePath)).then(() => { DO STH WITH tokenFilePath})
```
?